### PR TITLE
Added the option to decode a QR code

### PR DESCRIPTION
**Pull request by rincemust**
**Pull Request Label:**
* [ ] Bug
* [ ] New feature
* [x] Enhancement
* [ ] New component
* [ ] Typo

**Pull Request Checklist:**
- [x] Have you followed the [guidelines for contributing](https://github.com/alexanderepstein/Bash-Snippets/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/alexanderepstein/Bash-Snippets/pulls) for the same fix or component?
- [ ] Have you ran the tests locally with `bats tests`?

-----

The file must be of the PNG/JP(E)G/GIF format containing a QR code. The decoder is not compatible with wget, since according to the man page of wget, it cannot be used for multipart/form-data. The decoder has not been made for the 'fetch' client because I was unable to install the client.<br/><br/>

Decoding can be done by using `qrify -d filename.png`

### CODE REVIEW

File: qrify/qrify [hunk 1]
```diff
@@ -111,6 +111,35 @@ makeMultiLineQr()
   fi
 }
 
+# Function to get the json response from POST request
+function decodeQR {
```
Comment (retokromer) [inline qrify/qrify]: Why use the `function` syntax here?

File: qrify/qrify [hunk 2]
```diff
@@ -111,6 +111,35 @@ makeMultiLineQr()
...
+  if ! echo "$fileName" | grep -E -q ".*\.png$|.*\.PNG$|.*\.gif$|.*\.jpg$|.*\.jpeg$|.*\.GIF$|.*\.JPG$|.*\.JPEG$"
+  then
+    exit 1
+  fi
+
+  # only uses curl
+  # Cannot use wget because it does not support multipart/form-data (as per the man page)]
+
+  case "$configuredClient" in
+  curl) JSONresponse=$(curl -s -F "file=@$qrFile" http://api.qrserver.com/v1/read-qr-code/) || exit 1;;
```
Comment (retokromer) [inline qrify/qrify]: I suggest to indent (for readability).

File: qrify/qrify (hunk 1 above)
Comment (rincemust) [reply qrify/qrify]: There is no real reason, I guess... I mostly use this syntax. If there is an advantage in using the other syntax, please let me know!

File: qrify/qrify (hunk 2 above)
Comment (rincemust) [reply qrify/qrify]: Sure

File: qrify/qrify (hunk 1 above)
Comment (retokromer) [reply qrify/qrify]: The other parts of this package use it. And there is a mix of the two writings in this very script. (It‘s just a remark, nothing blocking at all.)

Comment (alexanderepstein) [reply qrify/qrify]: There is no advantage one way or another but I would definitely prefer if you didn't include the ```function``` syntax only to keep the entire script that same :smile:

File: qrify/qrify (hunk 2 above)
Comment (alexanderepstein) [reply qrify/qrify]: I agree, indenting is always good :+1:

**Comment by alexanderepstein** [comment]
Small little syntax stuff to fix but otherwise the addition to the script looks merge ready :tada:

**Pull request merged**